### PR TITLE
Fix messages from the extension being potentially missed

### PR DESCRIPTION
**Pull request by tomaka**
This PR refactors `extension.ts` in order to guarantee that messages from the extension concerning a chain can never be missed.

Furthermore, we no longer return an error at initialization if the extension sends an unexpected message. Instead, we just print a warning and move on. This is less harmful than introducing a state mismatch between the connect package and the extension.

### CODE REVIEW

File: packages/connect/src/connector/extension.ts [hunk 1]
```diff
@@ -57,95 +50,204 @@ export const createScClient = (): ScClient => {
...
-          ...msg,
-          origin: "substrate-connect-client",
-          chainId,
-        },
-        "*",
-      )
-    }
+    type ChainState =
+      | {
+          state: "wait-ack-or-err"
```
Comment (wirednkod) [inline packages/connect/src/connector/extension.ts:55]: should we name this state: "pending"? I mean - it does not really matter - just sounds better than "wait-ack-or-err"

File: packages/connect/src/connector/extension.ts [hunk 2]
```diff
@@ -57,95 +50,204 @@ export const createScClient = (): ScClient => {
...
+          jsonRpcMessage,
+        })
       },
       remove: () => {
-        if (crashError) throw crashError
-        if (!chains.has(chain)) throw new AlreadyDestroyedError()
-        listeners.delete(chainId)
+        if (chainState.state.state === "dead") {
+          throw chainState.state.error
+        }
```
Comment (wirednkod) [inline packages/connect/src/connector/extension.ts:230]: This code is multiple times, maybe we should add it in a function that will accept the state as an input?

**Review by wirednkod — APPROVED** [review]
Other than couple of nip tucks, lgtm

File: packages/connect/src/connector/extension.ts (hunk 2 above)
Comment (tomaka) [reply packages/connect/src/connector/extension.ts:230]: I mean, it's 3 lines of code that are present 3 times.

Comment (wirednkod) [reply packages/connect/src/connector/extension.ts:230]: Its not important. Just a proposal

**Pull request merged**